src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1625,7 +1625,11 @@ private AuthenticationResult refreshToken(final CallbackHandler callbackHandle,
                 // remove item from cache to avoid same usage of
                 // refresh token in next acquireToken call
                 removeItemFromCache(refreshItem);
-                return acquireTokenLocalCall(callbackHandle, activity, useDialog, request);
+                if (!request.isSilent() && callbackHandle.callback != null && activity != null) {
+                    return acquireTokenLocalCall(callbackHandle, activity, useDialog, request);
+                } else {
+                    throw new AuthenticationException(ADALError.AUTH_FAILED_SERVER_ERROR, request.getLogInfo() + errLogInfo);
```

Comment:
Why not using callback to send the error back to caller instead of throwing AuthenticationException?